libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2240,6 +2240,13 @@ bool TypeChecker::visit(FunctionCallOptions const& _functionCallOptions)
 		m_errorReporter.fatalTypeError(_functionCallOptions.location(), "Expected callable expression before call options.");
 		return false;
 	}
+
+	if (dynamic_cast<FunctionCallOptions const*>(&_functionCallOptions.expression()))
```

Comment:
This syntactic check will not cover al cases - you have to work with the expression type instead of the ast node.